Sources/Caching/DeviceCache.swift

```diff
@@ -274,17 +274,31 @@ class DeviceCache {
 
     // MARK: - attribution
 
-    func latestNetworkAndAdvertisingIdsSent(appUserID: String) -> [String: String] {
+    func latestAdvertisingIdsByNetworkSent(appUserID: String) -> [AttributionNetwork: String] {
         return self.userDefaults.read {
             let key = CacheKeyBases.attributionDataDefaults + appUserID
-            let latestNetworkAndAdvertisingIdsSent = $0.object(forKey: key) as? [String: String] ?? [:]
-            return latestNetworkAndAdvertisingIdsSent
+            let latestAdvertisingIdsByRawNetworkSent = $0.object(forKey: key) as? [String: String] ?? [:]
+
+            // convert keys from UserDefault from Integer String to AttributionNetwork
+            let latestSent: [AttributionNetwork: String] =
+                 latestAdvertisingIdsByRawNetworkSent.compactMapKeys { network in
+                     guard let networkRawValue = Int(network),
+                        let attributionNetwork = AttributionNetwork(rawValue: networkRawValue) else {
+                             Logger.error(Strings.attribution.latest_attribution_sent_user_defaults_invalid)
```

Comment:
I would include the `network` string in the error, so if we get a report with this message we can figure out why.